java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaMetadata.java

```diff
@@ -300,15 +336,15 @@ private boolean validPrefix(final String input) {
          * @param input string value to be tested
          * @return true if the string contains illegal characters, false otherwise.
          */
-        private boolean hasIllegalChars(final String input) {
+        private boolean hasIllegalKeyChars(final String input) {
             final char[] chars = input.toCharArray();
 
             for (final char c : chars) {
                 if (isControlCharacter(c)) {
                     return true;
                 }
 
-                for (char illegalKeyChar : ILLEGAL_KEY_CHARS) {
+                for (final char illegalKeyChar : ILLEGAL_KEY_CHARS) {
```

Comment:
When writing this code, since this is in the hot path, I should have made the contents of `ILLEGAL_KEY_CHARS` sorted and used `Arrays.binarySearch()` to determine if a character is present in the values.

It might not matter and might be just a micro optimization.